### PR TITLE
Fix net_unittests linking issue for Android

### PR DESCRIPTION
**Pull request by sherryzy**
b/332352428

### CODE REVIEW

File: net/BUILD.gn [hunk 1]
```diff
@@ -1405,7 +1405,7 @@ component("net") {
     sources += [ "disk_cache/blockfile/mapped_file_posix.cc" ]
   }
 
-  if (false) {
+  if (is_android) {
```
Comment (johnxwork) [inline net/BUILD.gn:1408]: I thought this is not true for us and we'd use is_starboard?

Comment (sherryzy) [reply net/BUILD.gn:1408]: I'm not sure why, but this change does make the linking issue go away?

Comment (andrewsavage1) [reply net/BUILD.gn:1408]: Upstream: https://source.chromium.org/chromium/chromium/src/+/main:net/BUILD.gn;l=1318-1323?q=net%2FBUILD.gn&ss=chromium%2Fchromium%2Fsrc. Looks to me like we should bring back that conditional and add `|| is_starboard` and rely on the NOTIMPLEMENTED here: https://source.chromium.org/chromium/chromium/src/+/main:net/base/network_change_notifier_passive.cc;l=122

Comment (andrewsavage1) [reply net/BUILD.gn:1408]: We want to add `|| is_starboard` as there should not be platform differences in the cobalt layer of the build, i.e. windows should use these same functions to link when it is built

**Pull request merged**